tiledb/sm/query/readers/result_tile.h

```diff
@@ -71,12 +71,72 @@ class Subarray;
 class ResultTile {
  public:
   /**
-   * For each fixed-sized attributes, the second tile in the tuple is ignored.
-   * For var-sized attributes, the first tile is the offsets tile and the second
-   * tile is the var-sized values tile. If the attribute is nullable, the third
-   * tile contains the validity vector.
+   * Class definition for the tile tuple.
    */
-  typedef tuple<Tile, Tile, Tile> TileTuple;
+  class TileTuple {
+   public:
+    /* ********************************* */
+    /*     CONSTRUCTORS & DESTRUCTORS    */
+    /* ********************************* */
+
+    /** Default constructor. */
+    TileTuple() = delete;
+
+    /** Constructor with var_size and nullable parameters. */
+    TileTuple(bool var_size, bool nullable) {
+      if (var_size) {
+        var_tile_ = Tile();
+      }
+
+      if (nullable) {
+        validity_tile_ = Tile();
+      }
+    }
+
+    /* ********************************* */
+    /*                API                */
+    /* ********************************* */
+
+    /** @returns Fixed tile. */
+    Tile& fixed_tile() {
+      return fixed_tile_;
+    }
+
+    /** @returns Var tile. */
+    Tile& var_tile() {
+      return var_tile_.value();
+    }
+
+    /** @returns Validity tile. */
+    Tile& validity_tile() {
+      return validity_tile_.value();
+    }
+
+    /** @returns Fixed tile. */
+    const Tile& fixed_tile() const {
+      return fixed_tile_;
+    }
+
+    /** @returns Var tile. */
+    const Tile& var_tile() const {
+      return var_tile_.value();
```

Comment:
`optional::operator*()` gives unchecked access to the underlying value. This could be used to implement `var_tile_unchecked()`. The disadvantage of the unchecked version is that asking for a value that's not there is undefined behavior, so care needs to be taken in calling.

Reply:
For now, the extra check doesn't seem to be impacting performance. I have been bitten by optional::operator*() a couple times so unless there's a performance gain, I'll keep .value() for now :)